core/admin.py

```diff
@@ -65,11 +67,19 @@ def algorithm(self, obj):
     def dataset(self, obj):
         return obj.algorithm_job.dataset
 
+    # how to combine the function?
     def log_preview(self, obj):
```

Comment:
This function can be a module level function.  We don't use `self` within the function, so within each AdminModel there would be a much shorter function (e.g.,
```
  def log_preview(self, obj):
    return _text_preview(obj.log)
```
and then the module level would be renamed to `_text_preview` and take the FileField object rather than the model.